modules/navigation/3d/nav_mesh_generator_3d.cpp

```diff
@@ -291,10 +314,7 @@ void NavMeshGenerator3D::generator_bake_from_source_geometry_data(Ref<Navigation
 	rcPolyMeshDetail *detail_mesh = nullptr;
 	rcContext ctx;
 
-	// added to keep track of steps, no functionality right now
-	String bake_state = "";
-
-	bake_state = "Setting up Configuration..."; // step #1
+	p_generator_task->bake_state_msg = "Setting up Configuration..."; // step #1
```

Comment:
This is kind of wasteful, as you probably doing allocations here for not good reasons. could you instead have thread assign an Enum , and then GUI can query the enum, and to display it have a static function that give the correpsonding string ?

Reply:
Theoretically yes but it is not worth it. Too much hassle and creates unwanted dependencies with module specific values. That entire bake function is a non-stop allocation in itself so in the grand scheme of things this does not matter.

Reply:
I meant exactly what you did ! 👍